Shared/Actions/Rich Menu/CreateMenuItem.swift

```diff
@@ -0,0 +1,255 @@
+        A Background for your Icon
+
+        Use this in combination with Background Shape to show a background behind your icon
+        """,
+        default: .default
+    )
+    var background: RMStyle
+
+    // SF Symbol
+    @Parameter(
+        title: "SF Symbol Name",
+        description: """
+        The Name of a SF Symbol
+
+        For available Symbols see Apple's Website (https://developer.apple.com/sf-symbols/) ])
+        """,
+        default: "plus"
+    )
+    var systemName: String
+
+    @Parameter(
+        title: "Foreground",
+        description: "The Color for your SF Symbol",
+        default: .default
+    )
+    var foreground: RMStyle
+
+    // Emoji
+    @Parameter(
+        title: "Emoji",
+        description: """
+        Any Emoji 😀.
+
+        Tap the Emoji button on your keyboard and select one emoji.
+        """,
+        default: "😀",
+        inputOptions: .init(keyboardType: .default)
+    )
+    var emoji: String
+
+    @Parameter(
+        title: "Data"
+    )
+    var data: String?
+
+    @Parameter(
+        title: "Background Style",
+        description: """
+        The style of the icon's background.
+        """,
+        default: .circle
+    )
+    var backgroundShape: RMBackgroundShape
+
+    func makeIcon() async -> Data? {
+        switch iconType {
+        case .sfSymbol:
+            return await RMIconContainer(
+                sfSymbol: systemName,
+                foregroundColor: foreground.color(),
+                backgroundColor: background.color(isBackground: true),
+                backgroundShape: backgroundShape
+            )
+            .render()
+        case .emoji:
+            return await RMIconContainer(
+                emoji: emoji,
+                backgroundColor: background.color(isBackground: true),
+                backgroundShape: backgroundShape
+            )
+            .render()
+        }
+    }
+
+    func perform() async throws -> some IntentResult & ReturnsValue<MenuItem>  {
+        let icon = await makeIcon()
+
+        let item = MenuItem(
+            title: menuTitle,
+            subtitle: subtitle,
+            icon: icon
+        )
+
+        return .result(value: item)
+    }
+}
+
+struct MenuItem: TransientAppEntity {
+    init() {
+        self.init(title: nil, subtitle: nil)
+    }
+
+    static var typeDisplayRepresentation: TypeDisplayRepresentation {
+        "Menu Item"
+    }
+
+    var displayRepresentation: DisplayRepresentation {
+        let title: LocalizedStringResource = "\(title ?? "")"
+        let subtitle: LocalizedStringResource? = if let subtitle  { "\(subtitle)" } else { nil }
+        let image: DisplayRepresentation.Image? = if let icon {
+            if #available(iOS 17.0, macOS 14.0, *) {
+                .init(data: icon.data, displayStyle: .default)
+            } else {
+                .init(data: icon.data)
+            }
+        } else {
+            nil
+        }
+
+        return if let subtitle {
```

Comment:
You don't need the conditional here since the `subtitle` parameter accepts `nil`.